llpc/util/llpcElfWriter.cpp

```diff
@@ -1164,7 +1164,7 @@ Result ElfWriter<Elf>::LinkGraphicsRelocatableElf(
 
 
         MergeSection(pTextSection3,
-                     alignTo(pTextSection1->secHead.sh_size, 0x10),
+                     alignTo(pTextSection1->secHead.sh_size, 0x4),
```

Comment:
This raises an interesting question. LLVM is able to align basic blocks for loops, and that can in fact have a performance benefit. Technically, up to 64 bytes of alignment can in some cases help. 256 bytes of alignment are only required for shader entry points.

Perhaps LLVM could emit a meaningful alignment for .text sections, and then this code could take that alignment into account when combining sections?

Reply:
Yes, that would be a good idea.  I don't know what the alignment should be.  For now, I lowered it to remove the number of NOPs to make debugging easier.